neuralhydrology/modelzoo/transformer.py

```diff
@@ -0,0 +1,167 @@
+import logging
+from typing import Dict
+import numpy as np
+import math
+
+import torch
+import torch.nn as nn
+
+from neuralhydrology.modelzoo.head import get_head
+from neuralhydrology.modelzoo.basemodel import BaseModel
+
+LOGGER = logging.getLogger(__name__)
+
+class Transformer(BaseModel):
+    """Transformer model class, which relies on PyTorch's TransformerEncoder class.
+
+    This class implements the encoder of a transformer network with a regression or probabilistic head. 
+
+    The model configuration is specified in the config file using the following options:
+    -- transformer_embedding_dimension : int representing the dimension of the input embedding space. 
```

Comment:
```suggestion
    -- transformer_embedding_dimension : integer representing the dimension of the input embedding space. 
```

To stay consistent with your naming below. Other option is to switch both to `int`. However, since you are not referring directly to Python classes here, I would use the full word (i.e., integer)